phpBB/assets/javascript/installer.js

```diff
@@ -339,6 +340,57 @@
 	}
 
 	/**
+	 * Processes status data
+	 *
+	 * @param status
+	 */
+	function processTimeoutResponse(status) {
+		if (statusCount === 12) { // 1 minute hard cap
+			status = 'fail';
+		}
+
+		if (status === 'continue') {
+			refreshRequested = false;
+			doRefresh();
+		} else if (status === 'running') {
+			statusCount++;
+			$('#loading_indicator').css('display', 'block');
+			setTimeout(queryInstallerStatus, 5000);
+		} else {
+			$('#loading_indicator').css('display', 'none');
+			addMessage('error',
+				[{
+					title: installLang.title,
+					description: installLang.msg
```

Comment:
Why, does `addMessage()` not like it if you pass in extra options? I would have thought that it would just ignore them, in which case it would be fine.